recipes/libnice/all/conanfile.py

```diff
@@ -0,0 +1,111 @@
+import os
+from conan import ConanFile
+from conan.tools.meson import Meson, MesonToolchain
+from conan.tools.files import copy, get, rmdir, rename, chdir, replace_in_file
+from conan.tools.layout import basic_layout
+from conan.tools.gnu import PkgConfigDeps
+from conan.errors import ConanInvalidConfiguration
+
+
+class LibniceConan(ConanFile):
+    name = "libnice"
+    homepage = "https://libnice.freedesktop.org/"
+    license = ("MPL-1.1", "LGPL-2.1-only")
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "a GLib ICE implementation"
+    topics = ("ice", "stun", "turn")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "crypto_library": ["openssl", "win32"],
+        "with_gstreamer": [True, False],
+        "with_gtk_doc": [True, False],
+        "with_introspection": [True, False]}
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_gstreamer": False,
+        "with_gtk_doc": False,
+        "with_introspection": False}
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            self.options.crypto_library = "win32"
+        else:
+            self.options.crypto_library = "openssl"
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def validate(self):
+        if self.settings.os != "Windows" and self.options.crypto_library == "win32":
+            raise ConanInvalidConfiguration(
+                "crypto_library=win32 is not supported on non-Windows")
+        if self.settings.os == "Windows" and self.options.with_gtk_doc:
+            raise ConanInvalidConfiguration(
+                "gtk-doc is disabled by libnice while building on Windows")
+        if self.settings.get_safe("runtime") in ("MT", "MTd", "static") and self.options["glib"].shared:
```

Comment:
```suggestion
        if is_msvc_static_runtime(self) and self.options["glib"].shared:
```
New safer way to check it: https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html#conan-tools-microsoft-is-msvc-static-runtime

Also, runtime is part of compiler, so correct approach would be `self.settings.get_safe("compiler.runtime")`